### ISSUE TITLE
Don’t demand quoting coproc FD numbers

### ISSUE THREAD

**Issue by lucaswerkmeister** [koalaman/shellcheck#858]
#### For bugs
- Rule Id (if any, e.g. SC1000): **SC2086**
- My shellcheck version (`shellcheck --version` or "online"): **online**
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```bash

#!/bin/bash
coproc cat
printf 'hello\n' >&${COPROC[1]}
read -r -u ${COPROC[0]} greeting
printf '%s\n' "$greeting"
kill %1

```

#### Here's what shellcheck currently says:

> ^-- SC2086: Double quote to prevent globbing and word splitting.

#### Here's what I wanted or expected to see:

Coproc arrays only contain file descriptor numbers, so quoting them is unnecessary. Since shellcheck already tracks coproc variables afaik, the quoting warning should be disabled for those variables.
